corehq/ex-submodules/casexml/apps/case/xform.py

```diff
@@ -140,6 +145,13 @@ def _get_or_update_cases(xforms, case_db):
     )
 
 
+def _validate_case_ids(case_ids):
+    for case_id in case_ids:
+        if case_id and not re.match(r'^[\w\-]*$', case_id):
+            raise InvalidCaseId("Invalid case id '%s'. Case id can have only alphanumeric characters, "
+                                "underscore and dash" % case_id)
```

Comment:
Do we need to support case ids with underscores? I don't remember ever seeing that. I'd prefer to support alphanumeric characters and dash only.

Reply:
I think Simon suggested that because that is the set supported by the url, so its safer that way.
I have not seen an underscore in a case id as well though. @snopoke thoughts?

Reply:
I just changed the wording of the comment. The regex is still the same and include `_` in the `\w` character class. Fine with me to exclude underscores.